### ISSUE TITLE
Support for Rollup / Vite

### ISSUE THREAD

**Issue by FlorianRappl** [smapiot/piral#435]
# New Feature Proposal

For more information, see the `CONTRIBUTING` guide.

## Description

Due to popular demand we should provide all means for yet another bundler in form of Rollup.js. This should also allow Vite to be used. Not sure yet if we need to distinguish or if it makes sense to have `piral-cli-rollup` and `piral-cli-vite`, but something like this may make sense.

## Background

I think for this it would be helpful to only support pilets v2, but we still would need a proper HTML entry plugin / way (may be similar to what we did for esbuild) and a codegen plugin (I think this one we already have working - just needs to be published).

## Discussion

Anything in particular that we need to respect?

**Comment by StJohn3D** [comment]
I'm new to Rollup and Vite but very interested. I've started using Vite for my latest projects and really like it. Regarding the question posed in the issue description. I just think that it is imported to avoid overly opinionated systems like Create React App. Best to allow people to customize their Rollup/Vite configs as much as possible.

**Comment by FlorianRappl** [comment]
> Best to allow people to customize their Rollup/Vite configs as much as possible.

Yeah fully agreed. Right now our way is to make every bundler work without a config (at least for "simple" / "initial" scenarios), but then allow everything to be configured. This way its easy to get started with, but not (too) opinionated and easy / open for customization.
